xbmc/input/linux/LIRC.cpp

```diff
@@ -109,109 +101,55 @@ void CRemoteControl::Disconnect()
-    Initialize();
-  }
 }
 
 void CRemoteControl::Initialize()
 {
-  struct sockaddr_un addr;
-  unsigned int now = XbmcThreads::SystemClockMillis();
-
-  if (m_bInitialized || !m_used || (now - m_lastInitAttempt) < (unsigned int)m_initRetryPeriod)
+  if (m_bInitialized || !m_used || IsRunning())
     return;
-  
-  m_lastInitAttempt = now;
+
+  Create();
+}
+void CRemoteControl::Process()
+{
+  struct sockaddr_un addr;
   addr.sun_family = AF_UNIX;
   strcpy(addr.sun_path, m_deviceName.c_str());
 
   CLog::Log(LOGINFO, "LIRC %s: using: %s", __FUNCTION__, addr.sun_path);
 
-  // Open the socket from which we will receive the remote commands
-  if ((m_fd = socket(AF_UNIX, SOCK_STREAM, 0)) != -1)
+  int iAttempt = 0;
+  unsigned int iMsRetryDelay = 5000;
+  unsigned int time = XbmcThreads::SystemClockMillis() - iMsRetryDelay;
+
+  // try to connect 60 times @ a 5 second interval (5 minutes)
+  // multiple tries because LIRC service might be up and running a little later then xbmc on boot.
+  while (!m_bStop && iAttempt <= 60)
   {
-    // Connect to the socket
-    if (connect(m_fd, (struct sockaddr *)&addr, sizeof(addr)) != -1)
-    {
-      int opts;
-      m_bLogConnectFailure = true;
-      if ((opts = fcntl(m_fd,F_GETFL)) != -1)
-      {
-        // Set the socket to non-blocking
-        opts = (opts | O_NONBLOCK);
-        if (fcntl(m_fd,F_SETFL,opts) != -1)
-        {
-          if ((m_file = fdopen(m_fd, "r+")) != NULL)
-          {
-#ifdef HAVE_INOTIFY
-            // Setup inotify so we can disconnect if lircd is restarted
-            if ((m_inotify_fd = inotify_init()) >= 0)
-            {
-              // Set the fd non-blocking
-              if ((opts = fcntl(m_inotify_fd, F_GETFL)) != -1)
-              {
-                opts |= O_NONBLOCK;
-                if (fcntl(m_inotify_fd, F_SETFL, opts) != -1)
-                {
-                  // Set an inotify watch on the lirc device
-                  if ((m_inotify_wd = inotify_add_watch(m_inotify_fd, m_deviceName.c_str(), IN_DELETE_SELF)) != -1)
-                  {
-                    m_bInitialized = true;
-                    CLog::Log(LOGINFO, "LIRC %s: successfully started", __FUNCTION__);
-                  }
-                  else
-                    CLog::Log(LOGDEBUG, "LIRC: Failed to initialize Inotify. LIRC device will not be monitored.");
-                }
-              }
-            }
-#else
-            m_bInitialized = true;
-            CLog::Log(LOGINFO, "LIRC %s: successfully started", __FUNCTION__);
-#endif
-          }
-          else
-            CLog::Log(LOGERROR, "LIRC %s: fdopen failed: %s", __FUNCTION__, strerror(errno));
-        }
-        else
-          CLog::Log(LOGERROR, "LIRC %s: fcntl(F_SETFL) failed: %s", __FUNCTION__, strerror(errno));
-      }
-      else
-        CLog::Log(LOGERROR, "LIRC %s: fcntl(F_GETFL) failed: %s", __FUNCTION__, strerror(errno));
-    }
-    else
+    if (XbmcThreads::SystemClockMillis() - time >= iMsRetryDelay)
```

Comment:
Can you use AbortableWait() as in https://github.com/garbear/xbmc/commit/65b96f2#diff-79dbb285338e176975ab76f60ad0a4e1R208?

This way we avoid additional context switches every second and don't hang for up to a second after Disconnect() is called